src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1600,11 +1600,12 @@ private IUsingOperation CreateBoundUsingStatementOperation(BoundUsingStatement b
         {
             Lazy<IOperation> resources = new Lazy<IOperation>(() => Create((BoundNode)boundUsingStatement.DeclarationsOpt ?? boundUsingStatement.ExpressionOpt));
             Lazy<IOperation> body = new Lazy<IOperation>(() => Create(boundUsingStatement.Body));
+            ImmutableArray<ILocalSymbol> locals = boundUsingStatement.Locals.CastArray<ILocalSymbol>();
```

Comment:
Use CastUp here instead of CastArray. CastUp will verify the cast semantics at runtime.